shogun-lib/src/main/java/de/terrestris/shogun/lib/controller/BaseController.java

```diff
@@ -472,10 +472,9 @@ public S updatePartial(@RequestBody Map<String, Object> values, @PathVariable("i
                 throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
             }
 
-            Optional<S> persistedEntity = service.findOne(entityId);
-
-            if (persistedEntity.isPresent()) {
-                S updatedEntity = service.updatePartial(entityId, values);
+            S persistedEntity = service.findOne(entityId).orElseThrow();
+            if (persistedEntity != null) {
+                S updatedEntity = service.updatePartial(entityId, persistedEntity, values);
```

Comment:
The update method includes the same check. Do we have to update this too?

Reply:
In `updatePartial` is checked whether the `entityId` matches the id of the persisted entity. Beside, the entity has to be passed to `updatePartial` otherwise the used target type in `hasPermission` is `Map<String, Object>` but it must be `S` (a sub-class of `BaseEntity`)